serial-settings/src/lib.rs

```diff
@@ -0,0 +1,306 @@
+    _menu: &menu::Menu<Context<Settings, Flash>>,
+    _item: &menu::Item<Context<Settings, Flash>>,
+    _args: &[&str],
+    _context: &mut Context<Settings, Flash>,
+) {
+    cortex_m::peripheral::SCB::sys_reset();
+}
+
+fn handle_reset<Settings: SerialSettings, Flash: NorFlash + 'static>(
+    _menu: &menu::Menu<Context<Settings, Flash>>,
+    _item: &menu::Item<Context<Settings, Flash>>,
+    _args: &[&str],
+    context: &mut Context<Settings, Flash>,
+) {
+    context.settings.reset();
+    context.save();
+}
+
+fn handle_get<Settings: SerialSettings, Flash: NorFlash + 'static>(
+    _menu: &menu::Menu<Context<Settings, Flash>>,
+    item: &menu::Item<Context<Settings, Flash>>,
+    args: &[&str],
+    context: &mut Context<Settings, Flash>,
+) {
+    let mut buf = [0u8; 256];
+    let key = menu::argument_finder(item, args, "item").unwrap().unwrap();
+    let len = match context.settings.get_json(key, &mut buf) {
+        Err(e) => {
+            writeln!(context, "Failed to read {key}: {e}").unwrap();
+            return;
+        }
+        Ok(len) => len,
+    };
+
+    let stringified = core::str::from_utf8(&buf[..len]).unwrap();
+    writeln!(context, "{key}: {stringified}").unwrap();
+}
+
+fn handle_set<Settings: SerialSettings, Flash: NorFlash + 'static>(
+    _menu: &menu::Menu<Context<Settings, Flash>>,
+    item: &menu::Item<Context<Settings, Flash>>,
+    args: &[&str],
+    context: &mut Context<Settings, Flash>,
+) {
+    let key = menu::argument_finder(item, args, "item").unwrap().unwrap();
+    let value = menu::argument_finder(item, args, "value").unwrap().unwrap();
+
+    // Now, write the new value into memory.
+    // TODO: Validate it first?
+    match context.settings.set_json(key, value.as_bytes()) {
+        Ok(_) => {
+            context.save();
+            writeln!(
+                context,
+                "Settings in memory may differ from currently operating settings. \
+        Reset device to apply settings."
+            )
+    .unwrap();
+        }
+        Err(e) => {
+            writeln!(context, "Failed to update {key}: {e:?}").unwrap();
+        }
+    }
+}
+
+pub struct SerialTerminal<'a, UsbBus: usb_device::bus::UsbBus, Settings: SerialSettings, Flash: NorFlash + 'static> {
+    usb_device: usb_device::device::UsbDevice<'a, UsbBus>,
+    usb_serial: usbd_serial::SerialPort<'a, UsbBus>,
+    menu: menu::Runner<'a, Context<Settings, Flash>>,
+    output: bbqueue::Consumer<'static, 512>,
+}
+
+impl<'a, UsbBus: usb_device::bus::UsbBus, Settings: SerialSettings, Flash: NorFlash + 'static> SerialTerminal<'a, UsbBus, Settings, Flash> {
+    pub fn new(
+        usb_device: usb_device::device::UsbDevice<'static, UsbBus>,
+        usb_serial: usbd_serial::SerialPort<'static, UsbBus>,
+        input_buffer: &'a mut [u8],
+        settings: Settings,
+        flash: Flash,
+    ) -> Self {
+        let (producer, consumer) = OUTPUT_BUFFER.try_split().unwrap();
```

Comment:
I suspect we can manage buffering without bbqueue, since there's synchronous access to the various objects.